### ISSUE TITLE
Eventually method doesn't handle parameters properly

### ISSUE THREAD

**Issue by joceron** [agourlay/cornichon#409]
I'm not sure if this is a bug or a question, but when I have this:
```
def testMethod(foo: Step): Step = Eventually(maxDuration = 15.seconds, interval = 200.milliseconds) {
  foo
}
```
The compiler throws this error:
```
error: Unsupported expression. Only expressions of type `com.github.agourlay.cornichon.core.Step` are allowed here.
found 'foo' of type 'com.github.agourlay.cornichon.core.Step'
```

But it does compile if I do this:
```
def testMethod(foo: Step): Step = Eventually(maxDuration = 15.seconds, interval = 200.milliseconds) {
  dummyStep
}

def dummyStep: Step = When I print_step("")
```
And it also compiles if I do this:
```
def testMethod(foo: Step): Step = Eventually(maxDuration = 15.seconds, interval = 200.milliseconds) {
  dummyStep
  foo
}

def dummyStep: Step = When I print_step("")
```

I also tried using `foo: => Step` as an argument, but I get the same result

Here would be a whole, simple file with all the content:
```
import com.github.agourlay.cornichon.CornichonFeature
import com.github.agourlay.cornichon.core.Step

import scala.concurrent.duration._

trait TestTrait extends CornichonFeature {

  // This works
  def workingMethod(unusedStep: Step): Step = Eventually(maxDuration = 15.seconds, interval = 200.milliseconds) {
    dummyStep
  }
  def dummyStep: Step = When I print_step("")

  // This doesn't work
  def failingMethod(usedStep: Step): Step = Eventually(maxDuration = 15.seconds, interval = 200.milliseconds) {
    usedStep
  }
}

```

**Comment by agourlay** [comment]
Thank you very much for the very detailed issue 👍 

This is indeed a bug in the library, the macro collecting the steps does not handle this special case.

I have pushed a fix with https://github.com/agourlay/cornichon/pull/410 and I will try to cut a release for it as it has been a while since `0.19.4`

**Comment by agourlay** [comment]
fixed in `0.19.5` https://github.com/agourlay/cornichon/releases/tag/v0.19.5
